credsweeper/rules/rule.py

```diff
@@ -69,14 +63,14 @@ def __init__(self, config: Config, rule_dict: Dict) -> None:
             self.__rule_type: RuleType = rule_type
         else:
             self._malformed_rule_error(rule_dict, Rule.TYPE)
-        self.__patterns = Rule._get_patterns(self.rule_type, rule_dict[Rule.VALUES])
+        self.__patterns: List[re.Pattern] = []
+        self._init_patterns(rule_dict[Rule.VALUES])
```

Comment:
What about move `self.__patterns` to `_init_patterns()`?
It can make reduce the lines of `__init__()`.

Reply:
I scare, some of development tools are not properly analyse the code where a variable is defined (and typed) ontside __init__.
That why i keep this approach

Reply:
Okay..
But as you know previous approach of initializing the variables was `get_xxx()` and it can reduce the code size and more easy to read the code.
Why you changed the methods?

Reply:
Malformed rule exception and warnings might be raised with the methods. To point which rule is fail we've got  ``self.rule_name`` already. Also self.config is used.
I decided reduce complexity of arguments for the private methods - only configuration is passed.
Suppose, a return is extra for "self" method when it may be set right in the method.

Reply:
Yup I know about the `self.rule_name`.
And IMO below code block is more readable and can reduce the LOC.. isn't it?

```suggestion
        self.__patterns: List[re.Pattern] = self._init_patterns(rule_dict[Rule.VALUES])
```

Reply:
refactoring was done

Reply:
Thank you for fix! I checked.